### PR TITLE
#4198 save site data on vaccine dispense

### LINKED ISSUE #4198: When saving vaccination site data - store in [transact]custom_data as an object
Child of #4176 
 
## Is your feature request related to a problem? Please describe.
Tonga (and Kiribati??) want #4176 and this is a step required to achieve it.

## Describe the solution you'd like
See #4176

## Implementation
- When '`Confirm`' (or '`OK & Next`') are selected at the end of the vaccine dispensing flow (currently the screen where the vaccine is selected), ensure that the data entered in the form is stored in `[transact]custom_data as an object`

## Describe alternatives you've considered
Making someone unhappy? Maybe this didn't even need to be a sub-issue 🤔  can close two issues with one PR if needed though.

## Additional context
N/A

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #4198

## Change summary

- Adds `Transaction.customData` to mobile schema/sync utils
- Save siteData into `Transaction.customData` when creating a vaccinePrescription if it exists

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Create a prescription with siteData (needs FormSchema with type `VaccineSite`), check Realm Explorer (and/or export and check the contents that way) and verify that the siteData is saved to the transaction
- [ ] Sync to Desktop - check via record browser that [transact].custom_data contains the site data for the vaccine prescription just created 

### Related areas to think about
Dispensing without siteData (and regular customer invoices) should not create any `customData` for the transactions

### CODE REVIEW

**Comment by katherine-sussol** [comment]
Oops something weird happened with my diff 🙀

File: src/database/utilities/createRecord.js [hunk 1]
```diff
@@ -566,7 +566,7 @@ const createCustomerRefundLine = (database, customerCredit, transactionBatch) =>
  * @param   {Name}         customer  Customer associated with invoice.
  * @return  {Transaction}
  */
-const createCustomerInvoice = (database, customer, user, mode = 'store') => {
+const createCustomerInvoice = (database, customer, user, mode = 'store', customData = null) => {
```
Comment (joshxg) [inline src/database/utilities/createRecord.js:569]: `JSON.stringify(null) === 'null'` - maybe not what we want? think just `''` when empty?

**Review by joshxg — APPROVED** [review]
sweet

File: src/database/utilities/createRecord.js (hunk 1 above)
Comment (katherine-sussol) [reply src/database/utilities/createRecord.js:569]: Eep, thanks

Comment (katherine-sussol) [reply src/database/utilities/createRecord.js]: Did a bit more testing, think it's better to set the record value to `null` if not provided (for consistency with `parseJSONData` in `incomingSyncUtils`) so transactions without `customData` in mobile are consistent with those synced from Desktop 🤣  
(prev impl had records from Desktop with `null` and those from Mobile with `''` - oops)

Soz bit late response as well, too much fun testing different JSON schemas out 😹 
25641e3

Comment (joshxg) [reply src/database/utilities/createRecord.js]: Ah, good points!
